### PR TITLE
Cookbook: Regexp (ppx_regexp)

### PR DESCRIPTION
**Pull request by F-Loyer**
It was quite logic to cover the Str regexp since Str is shipped with OCaml. But ppx_regexp (with Re) makes some things far easier.

### CODE REVIEW

**Review by F-Loyer — COMMENTED** [review]
Comment fixed

File: data/cookbook/text-processing/regular-expressions-ppxregexp/00-ppx_regexp.ml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+- name: "re"
+  version: "1.10.4"
+- name: "ppx_regexp"
+  version: "0.5.1"
+libraries:
+  - re
+ppxes:
+  - ppx_regexp
+discussion: |
+  - **Understanding `re`:** The `Re` library propose multiple advantage over the `Str` library which is shipped with OCaml. It supports multiple syntaxes, and its absence of global states permits concurrent parttern matching. It is completed by the `ppx_regexp` which makes the use of this library easier. However, only the PCRE syntax is supported.
```
Comment (christinerose) [inline data/cookbook/text-processing/regular-expressions-ppxregexp/00-ppx_regexp.ml]:
```suggestion
  - **Understanding `re`:** The `re` library proposes multiple advantages over the `Str` library, which is shipped with OCaml. It supports multiple syntaxes, and its absence of global states permits concurrent pattern matching. It is completed by the `ppx_regexp`, which makes using this library easier. However, only the PCRE syntax is supported.
```

File: data/cookbook/text-processing/regular-expressions-ppxregexp/00-ppx_regexp.ml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+packages:
+- name: "re"
+  version: "1.10.4"
+- name: "ppx_regexp"
+  version: "0.5.1"
+libraries:
+  - re
+ppxes:
```
Comment (christinerose) [inline data/cookbook/text-processing/regular-expressions-ppxregexp/00-ppx_regexp.ml:9]:
```suggestion
PPXs:
```

File: data/cookbook/text-processing/regular-expressions-ppxregexp/00-ppx_regexp.ml [hunk 3]
```diff
@@ -6,7 +6,7 @@ packages:
   version: "0.5.1"
 libraries:
   - re
-PPXs:
+ppxes:
```
Comment (christinerose) [inline data/cookbook/text-processing/regular-expressions-ppxregexp/00-ppx_regexp.ml:9]: is "ppxes" a specific command or are you talking about multiple PPX? If the latter, the style guide has this formatted as PPXs.

Comment (F-Loyer) [reply data/cookbook/text-processing/regular-expressions-ppxregexp/00-ppx_regexp.ml:9]: Here, `ppxes` is the token expected by [cookbook.ml](https://github.com/ocaml/ocaml.org/tree/cookbook/tool/ood-gen/lib/cookbook.ml#L25)

Comment (christinerose) [reply data/cookbook/text-processing/regular-expressions-ppxregexp/00-ppx_regexp.ml:9]: Thanks for clarifying!

**Comment by cuihtlauac** [comment]
Set upstream from https://github.com/ocaml/ocaml.org/tree/cookbook-regexp2

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
Set upstream from https://github.com/ocaml/ocaml.org/tree/cookbook-regexp2

**Pull request closed without merging**